qt-bugs@ issue : unknown
Trolltech task ID : unknown
applied: no
author: nolis71cu

this patch add to the Qftp class an overload get funtion with a giveing offset.
in order to restart a broken download.

===================================================================
*** src/network/qftp.h.orig	Thu Dec 29 09:11:18 2005
--- src/network/qftp.h	Sun Jan  1 23:11:29 2006
***************
*** 91,96 ****
--- 91,97 ----
      int list(const QString &dir = QString());
      int cd(const QString &dir);
      int get(const QString &file, QIODevice *dev=0, TransferType type = Binary);
+     int get(const QString &file, qulonglong offSet, QIODevice *dev=0, TransferType type = Binary);
      int put(const QByteArray &data, const QString &file, TransferType type = Binary);
      int put(QIODevice *dev, const QString &file, TransferType type = Binary);
      int remove(const QString &file);
*** src/network/qftp.cpp.orig	Thu Dec 29 09:11:18 2005
--- src/network/qftp.cpp	Sun Jan  1 23:10:39 2006
***************
*** 1802,1807 ****
--- 1802,1826 ----
  
  /*!
      \overload
+ get remote file starting at offSet
+ */
+ 
+ int QFtp::get(const QString &file, qulonglong offSet, QIODevice *dev, TransferType type)
+ {
+     QStringList cmds;
+     cmds << ("SIZE " + file + "\r\n");
+     if (type == Binary)
+         cmds << "TYPE I\r\n";
+     else
+         cmds << "TYPE A\r\n";
+     cmds << (d_func()->transferMode == Passive ? "PASV\r\n" : "PORT\r\n");
+     cmds << ("REST " + QString::number(offSet) + "\r\n");
+     cmds << ("RETR " + file + "\r\n");
+     return d_func()->addCommand(new QFtpCommand(Get, cmds, dev));
+ }
+ 
+ /*!
+     \overload
  
      Writes a copy of the given \a data to the file called \a file on
      the server. The progress of the upload is reported by the
